R/set_tech_trajectories.R

```diff
@@ -464,3 +458,38 @@ calc_late_sudden_traj <- function(start_year = 2020,
 
   return(late_and_sudden)
 }
+
+#' Remove negative late and sudden rows
+#'
+#' Function checks for negative values on variable late_and_sudden. All
+#' technology x company_name combinations holding >= 1 negative value are
+#' removed.
+#'
+#' @param data_with_late_and_sudden A tibble containing scenario data with
+#'   projected late and sudden trajectory.
+#'
+#' @return Input tibble with potentially removed rows.
+filter_negative_late_and_sudden <- function(data_with_late_and_sudden) {
+  negative_late_and_sudden <- data_with_late_and_sudden %>%
+    dplyr::filter(.data$late_sudden < 0) %>%
+    dplyr::select(.data$company_name, .data$technology) %>%
```

Comment:
why not save the select line and directly go `dplyr::distinct(.data$company_name, .data$technology)`? same result, less lines...